Refactor/navigation

Joseph-Sameh-0 opened the pull request:
Add bottom nav bar to TudeeNavGraph  for Home and Task and Category screen

Joseph-Sameh-0 committed:
refactor: Introduce MainNavGraph for bottom navigation

This commit introduces a new `MainNavGraph` to handle navigation between the main screens (Home, Task, Category) using a bottom navigation bar.

Key changes:
- Created `MainNavGraph.kt` to define the navigation logic for the main screens.
- Added `MainNav` to `Screens.kt` to represent the main navigation flow.
- Updated `TudeeNavGraph.kt` to navigate to `MainNav` from `OnBoardScreen`.
- Modified `TudeeNavigationBar.kt` to accept a list of `NavItem` and an `onItemClick` callback to handle navigation.
- `NavItem` now includes a `screen` property to link it to a specific `Screens` destination.
- Integrated `TudeeNavigationBar` into `MainNavGraph` to enable bottom navigation.
- Established `LocalMainNavController` to provide the `NavController` for the main navigation flow.

Joseph-Sameh-0 committed:
update navigation to TaskForm in HomeScreen

AzizaHelmy commented on app/src/main/java/com/example/tudeeapp/presentation/navigation/Screens.kt:
```diff
@@ -11,6 +11,9 @@ sealed class Screens {
     data object OnBoarding : Screens()
 
+    @Serializable
+    data object MainNav : Screens()
```
MainNav isn't a screen

AzizaHelmy commented on app/src/main/java/com/example/tudeeapp/presentation/navigation/MainNavGraph.kt:
```diff
@@ -0,0 +1,60 @@
+    compositionLocalOf<NavHostController> { error("No Nav Controller Found") }
+
+@Composable
+fun MainNavGraph() {
```
Agin for my opinion I don't prefer this approach in seperation graphs

MustafaIbrahim96 replied:
She said in the meeting that splitting the navigation is usually done in larger projects
But in our case, we have a small number of screens, so why would we split the navigation?
In my opinion, the splitting part has become a bit complicated. We can do it, but only after we fully understand the basic approach.

Joseph-Sameh-0 replied:
I'm working on it

Joseph-Sameh-0 committed:
refactor: Remove parameters from Home and Task screens

- Removed `userName` parameter from `HomeScreen` and its navigation.
- Removed `taskId` and `taskTitle` parameters from `TaskScreen` and its navigation.
- Updated `TudeeNavigationBar` to use object instances for Home and Task screens.
- Adjusted `MainNavGraph` to hide the bottom bar when navigating to screens other than Home, Task, and Category.

Joseph-Sameh-0 committed:
refactor: Move main navigation graph to TudeeNavGraph

The `MainNavGraph.kt` file has been deleted and its functionality has been integrated into `TudeeNavGraph.kt`. This centralizes the navigation logic.

The `Screens.MainNav` object has also been removed from `Screens.kt` as it's no longer needed.

OnBoardScreen now navigates to `Screens.Home` instead of the removed `Screens.MainNav`.

DinaOthman21 commented:
great work

Joseph-Sameh-0 committed:
Merge branch 'development' into refactor/navigation

Mohamed-Elhanafy committed:
remove snackbar from [TudeeScaffold].
